### ISSUE TITLE
PathToStrings API Issue with special characters in input path

### ISSUE THREAD

**Issue by schandrashek** [openconfig/ygot#597]
PathToStrings doesn't work as expected with paths having special characters.

Eg: /components/component[name=abc[123]]

Here, the string returned from the API is /components/component[name=abc[123\\]]. 

The issue here is in the elemToString() function, where only a ']' is correctly escaped and there's no escape for '['.

**Comment by wenovus** [comment]
I agree that it's inconsistent that, as in your referred commit, both `=` and `]` are escaped, despite the former being unambiguous, and `[` is not escaped even though it's also a special character. @robshakir Do you mind if I fix this inconsistency by escaping `[` as well and then adding the decision to the gNMI spec? I think I agree with Shachindra here.
